document-store/src/main/java/org/hypertrace/core/documentstore/metric/mongo/MongoDocStoreMetricProvider.java

```diff
@@ -0,0 +1,68 @@
+package org.hypertrace.core.documentstore.metric.mongo;
+
+import static org.hypertrace.core.documentstore.model.config.mongo.MongoDefaults.ADMIN_DATABASE;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoDatabase;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.bson.Document;
+import org.hypertrace.core.documentstore.metric.BaseDocStoreMetricProviderImpl;
+import org.hypertrace.core.documentstore.metric.DocStoreMetric;
+import org.hypertrace.core.documentstore.model.config.mongo.MongoConnectionConfig;
+import org.hypertrace.core.documentstore.mongo.MongoDatastore;
+
+@Slf4j
+public class MongoDocStoreMetricProvider extends BaseDocStoreMetricProviderImpl {
+  private static final String NUM_ACTIVE_CONNECTIONS_METRIC_NAME = "num.active.mongo.connections";
+  private static final String APP_NAME_LABEL = "app_name";
+
+  private final MongoDatabase adminDb;
+  private final String applicationNameInCurrentConnection;
+
+  public MongoDocStoreMetricProvider(
+      final MongoDatastore dataStore,
+      final MongoConnectionConfig connectionConfig,
+      final MongoClient client) {
+    super(dataStore);
+    adminDb = client.getDatabase(ADMIN_DATABASE);
+    applicationNameInCurrentConnection = connectionConfig.applicationName();
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public DocStoreMetric getConnectionCountMetric() {
+    try {
+      final Document document =
+          adminDb.runCommand(new Document("currentOp", 1)).append("$all", true);
+      final List<Map<String, Object>> processes = document.get("inprog", List.class);
+      long count = 0;
+
+      for (final Map<String, Object> process : processes) {
+        final String appName = process.getOrDefault("appName", "").toString();
+        final boolean isActive =
+            Boolean.parseBoolean(process.getOrDefault("active", "false").toString());
+
+        if (appName.equals(applicationNameInCurrentConnection) && isActive) {
+          count++;
+        }
+      }
+
+      final DocStoreMetric metric =
+          DocStoreMetric.builder()
+              .name(NUM_ACTIVE_CONNECTIONS_METRIC_NAME)
+              .value(count)
+              .labels(Map.of(APP_NAME_LABEL, applicationNameInCurrentConnection))
+              .build();
+      log.debug("Returned metric: {}", metric);
+      return metric;
+    } catch (final Exception e) {
+      log.error("Unable to capture {}", NUM_ACTIVE_CONNECTIONS_METRIC_NAME, e);
+      return DocStoreMetric.builder()
```

Comment:
This means in any error case, we are saying, there are `0` connections. But, we actually don't know are there no connection or there are `x` connection. How is miss handled in metrics world - in our case in Prometheus?

E.g with this impl, lets say client reports metrics every 1 min:
first call: x connection -> prometheus: x connection total_count = x
second call: 0 connection due to error -> prometheus: 0 connection total_count = x
third call: x connection -> prometheus : x connection total_count = 2x

Or, if we return `null`, and client doesn't report.
first call: x connection -> prometheus: x connection total_count = x
second call: null due to error  and not reported to prometheus vai client: prom exporter will still read the value x connection : total_count = x
third call : x connection -> prometheus x connection total_count = 2x

But, here there will be a flat line. And above case, it will be deep to zero. I think, both cases are handled in Prometheus for counter metrics.

So, I think, we should be fine here.